### PR TITLE
Camera stacking support for the 2D Renderer

### PR DESCRIPTION
**Pull request by rainsing**
### Purpose of this PR
Camera stacking support for the 2D Renderer.

---
### Testing status

**Manual Tests**: What did you do?
- [x] Opened test project + Run graphic tests locally
- [x] Built a player
- [ ] Checked new UI names with UX convention
- [ ] Tested UI multi-edition + Undo/Redo + Prefab overrides + Alignment in Preset
- [x] C# and shader warnings (supress shader cache to see them)
- [ ] Checked new resources path for the reloader (in developer mode, you have a button at end of resources that check the paths)
- Other: 

**Automated Tests**: 
Added a few runtime tests that cover render textures creation of Base Cameras.

---
### Comments to reviewers
N/A

### CODE REVIEW

File: com.unity.render-pipelines.universal/Runtime/2D/Renderer2D.cs [hunk 1]
```diff
@@ -29,14 +36,23 @@ public Renderer2D(Renderer2DData data) : base(data)
...
             m_UseDepthStencilBuffer = data.useDepthStencilBuffer;
 
-            m_AfterPostProcessColorHandle.Init("_AfterPostProcessTexture");
-            m_ColorGradingLutHandle.Init("_InternalGradingLut");
+            // We probably should declare these names in the base class,
+            // as they must be the same across all ScriptableRenderer types for camera stacking to work.
+            k_ColorTextureHandle.Init("_CameraColorTexture");
+            k_DepthTextureHandle.Init("_CameraDepthAttachment");
+            k_AfterPostProcessColorHandle.Init("_AfterPostProcessTexture");
+            k_ColorGradingLutHandle.Init("_InternalGradingLut");
```
Comment (rainsing) [inline com.unity.render-pipelines.universal/Runtime/2D/Renderer2D.cs:48]: As said in the comments here, @phi-lira do you think we should put these strings into the base class?

File: com.unity.render-pipelines.universal/Runtime/2D/Renderer2D.cs [hunk 2]
```diff
@@ -29,14 +36,23 @@ public Renderer2D(Renderer2DData data) : base(data)
...
 
             m_Renderer2DData = data;
+
+            supportedRenderingFeatures = new RenderingFeatures()
+            {
+                cameraStacking = true,
```
Comment (rainsing) [inline com.unity.render-pipelines.universal/Runtime/2D/Renderer2D.cs:54]: Now that the 2D Renderer supports camera stacking, we should change the manual to reflect the new reality. @phi-lira Can the doc people on your side do this? Or should I ask our tech writer?

File: com.unity.render-pipelines.universal/Runtime/2D/Renderer2D.cs (hunk 1 above)
Comment (phi-lira) [reply com.unity.render-pipelines.universal/Runtime/2D/Renderer2D.cs:48]: Yes, this + we were thinking about making pipeline textures accessible to user. Something like `UniversalRenderTextureType.CameraColorTexture`.

Then you could do something like `renderer.GetTexture(UniversalRenderTextureType.CameraColorTexture); `

Comment (phi-lira) [reply com.unity.render-pipelines.universal/Runtime/2D/Renderer2D.cs:48]: What do you think about this approach?

File: com.unity.render-pipelines.universal/Runtime/2D/Renderer2D.cs (hunk 2 above)
Comment (phi-lira) [reply com.unity.render-pipelines.universal/Runtime/2D/Renderer2D.cs:54]: Ideally your tech write but we can also help if needed.

File: com.unity.render-pipelines.universal/Runtime/2D/Renderer2D.cs [hunk 3]
```diff
@@ -49,104 +65,144 @@ public Renderer2DData GetRenderer2DData()
...
-            bool useOffscreenColorTexture =
-                ppcUsesOffscreenRT || postProcessEnabled || cameraData.isHdrEnabled || cameraData.isSceneViewCamera || !cameraData.isDefaultViewport || !m_UseDepthStencilBuffer;
-
-            // Pixel Perfect Camera may request a different RT size than camera VP size.
-            // In that case we need to modify cameraTargetDescriptor here so that all the passes would use the same size.
-            if (ppcUsesOffscreenRT)
+
+            if (cameraData.renderType == CameraRenderType.Base)
+            {
+                m_CreateColorTexture = forceCreateColorTexture
```
Comment (phi-lira) [inline com.unity.render-pipelines.universal/Runtime/2D/Renderer2D.cs:80]: We should probably add some helper function to check these flags like to create an intermediate render texture in future as well

**Comment by rainsing** [comment]
> Just wanted to note that the Yamato failures on metal need to be looked at. I think some of them are simply incorrect reference images while others look like real issues.

Thanks for the reminder! I found the issue that was causing the failures and will re-run Yamato after the fix.

File: com.unity.render-pipelines.universal/Runtime/2D/Renderer2D.cs (hunk 1 above)
Comment (rainsing) [reply com.unity.render-pipelines.universal/Runtime/2D/Renderer2D.cs:48]: Sounds good.

File: com.unity.render-pipelines.universal/Runtime/2D/Renderer2D.cs (hunk 2 above)
Comment (rainsing) [reply com.unity.render-pipelines.universal/Runtime/2D/Renderer2D.cs:54]: OK then I'll ask our tech writer to update the doc in another PR.

**Pull request merged**